client/state/plugins/wporg/reducer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import React from 'react/addons';
+
+function updatePluginState( state = {}, pluginSlug, attributes ) {
+	return Object.assign( {},
+		state,
+		{ [ pluginSlug ]: Object.assign( {}, state[ pluginSlug ], attributes ) }
+	);
+}
```

Comment:
Hm. Maybe we should push for Immutable.js, though I'm personally still not entirely sold on it. The other reason I didn't bring it up was that this PR is nascent and I didn't want to condition the ongoing work too much.